packages/core/core/src/worker-report.js

```diff
@@ -0,0 +1,43 @@
+// @flow strict-local
+import type {ReporterEvent} from '@parcel/types';
+import ThrowableDiagnostic, {errorToDiagnostic} from '@parcel/diagnostic';
+import {PluginLogger} from '@parcel/logger';
+import type {WorkerApi} from '@parcel/workers';
+
+import ParcelConfig from './ParcelConfig';
+import registerCoreWithSerializer from './registerCoreWithSerializer';
+import PluginOptions from './public/PluginOptions';
+import '@parcel/cache'; // register with serializer
+import '@parcel/package-manager';
+import '@parcel/fs';
+
+registerCoreWithSerializer();
+
+export async function runReport(
+  workerApi: WorkerApi,
+  {
+    config,
+    opts,
+    event,
+  }: {|
+    config: ParcelConfig,
```

Comment:
use a reference for this?
`let {ref, dispose} = await workerfarm.createSharedReference(sharedValue);`